pkg/restapi/operation/common/models.go

```diff
@@ -42,3 +42,63 @@ type CreatePublicDIDResponse struct {
 	// TODO return base64-encoded raw bytes of the DID doc [Issue: #855]
 	DID *did.Doc `json:"did"`
 }
+
+// RegisterMessageServiceRequest model
+//
+// This is used for operation to register a message service to message handler
+//
+// swagger:parameters registerMsgSvc
+type RegisterMessageServiceRequest struct {
+	// Params for registering message service
+	//
+	// in: body
+	Params *RegisterMsgSvcParams
+}
+
+// RegisterMsgSvcParams contains parameters for registering a message service to message handler
+type RegisterMsgSvcParams struct {
+	// Name of the message service
+	// required: true
+	Name string `json:"name"`
+
+	// Acceptance criteria for message service based on message purpose
+	// in case of multiple purposes, message will be dispatched if any one of the purpose matches
```

Comment:
*if any one of the purpose matches* @llorllale @sudeshrshetty @Moopli is this correct?

Reply:
Yes, if someone can register single handler for set of purposes he is interested in. Instead of registering one for each purpose which is too tiresome.

Reply:
I think I will introduce a priority mechanism, in which case we'll update this with priority ordering.